packages/libsql-client/src/sqlite3.ts

```diff
@@ -63,8 +63,20 @@ export function _createClient(config: ExpandedConfig): Client {
         }
     }
 
-    // note: we must always prepend file scheme in order for SQLite3 to recognize :memory: connection query parameters
-    const path = `${config.scheme}:${config.path}`;
+    let isInMemory = isInMemoryConfig(config);
+    if (isInMemory && config.syncUrl) {
+        throw new LibsqlError(
+            `Embedded replica must use file for local db but URI with in-memory mode were provided instead: ${config.path}`,
+            "URL_INVALID",
+        );
+    }
+
+    // libsql API for embedded replicas expect to have raw path file without any schemes (not URI) - so we should be careful with prepending schema to it
```

Comment:
Please wrap comment lines to something like 80-100 columns to make them readable.